File: backend-dotnet/src/Core/App.Template.Application/DTOs/Auth/LoginResponseDto.cs
namespace AppTemplate.Application.DTOs.Auth;

/// <summary>
/// Login response with JWT token
/// </summary>
public record LoginResponseDto
{
    /// <summary>
    /// JWT access token
    /// </summary>
    public string Token { get; init; } = string.Empty;

    /// <summary>
    /// Token type (usually "Bearer")
    /// </summary>
    public string TokenType { get; init; } = "Bearer";

    /// <summary>
    /// Token expiration time in seconds
    /// </summary>
    public int ExpiresIn { get; init; }

    /// <summary>
    /// Refresh token for obtaining new access tokens
    /// </summary>
    public string? RefreshToken { get; init; }

    /// <summary>
    /// Refresh token expiration time
    /// </summary>
    public DateTime? RefreshTokenExpiresAt { get; init; }

    /// <summary>
    /// User information
    /// </summary>
    public UserInfoDto? User { get; init; }
}

File: backend-nestjs/src/modules/auth/application/dto/login-response.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class LoginResponseDto {
  @ApiProperty({ description: 'JWT access token' })
  accessToken: string;

  @ApiProperty({ description: 'Refresh token' })
  refreshToken: string;

  @ApiProperty({ description: 'Token type', default: 'Bearer' })
  tokenType: string;

  @ApiProperty({ description: 'Access token expiration time in seconds' })
  expiresIn: number;

  constructor(
    accessToken: string,
    refreshToken: string,
    expiresIn: number,
    tokenType: string = 'Bearer',
  ) {
    this.accessToken = accessToken;
    this.refreshToken = refreshToken;
    this.tokenType = tokenType;
    this.expiresIn = expiresIn;
  }
}
